x/emissions/module/rewards/rewards.go

```diff
@@ -159,6 +121,44 @@ func EmitRewards(
 	return nil
 }
 
+// New function for processing each topic individually
+func processTopic(
```

Comment:
can we write a unit test for this

Reply:
also process topic is a generic name that isn't great at conveying meaning of what this function does

Reply:
```suggestion
func getDistributionAndPayoutRewardsToTopicActors(
```

maybe something like this, idk

Reply:
good calls

Reply:
pending unit test